cryptofeed/exchanges/binance.py

```diff
@@ -279,13 +279,19 @@ async def _snapshot(self, pair: str) -> None:
                     max_depth = d
                     break
 
+        req_time = time.time()
         resp = await self.http_conn.read(self.rest_endpoints[0].route('l2book', self.sandbox).format(pair, max_depth))
         resp = json.loads(resp, parse_float=Decimal)
+        if 'E' not in resp:
+            resp_time = time.time()
+            timestamp = req_time + (resp_time - req_time) / 2
```

Comment:
the timestamp field in the data object must always come from the exchange. you cannot generate a timestamp and put it in there. if the exchange does not provide a timestamp, it should be none, like the code currently is

Reply:
Ok -- resubmitted.